### DISCUSSION TITLE
blog/inject-function-the-right-way/

### DISCUSSION THREAD

**Discussion by giscus** [nartc/tnc-scully#49] [Announcements]
# blog/inject-function-the-right-way/

Personal blog by Chau Tran

https://nartc.me/blog/inject-function-the-right-way/

**Comment by JanTheMan1010** [comment]
I have a special class which doesn't know the context in the moment it's created (when constructor is called).

Later on the class needs to know the context and currently I'm doing something like this:

[a]----------
class SpecialClass {
myService: MyService;
myMethod(injector: Injector): void {
	this.myService = injector.get(MyService);
	this.myService.doSomething();
}
}
------------

Now I'm curious if there is any advantage of doing it this instead:

[b]----------
class SpecialClass {
myService: MyService;
myMethod(injector: Injector): void {
	runInInjectionContext(injector, () => {
		this.myService = inject(MyService);
	});
	this.myService.doSomething();
}
}
------------

This page is the only google hit for "runInInjectionContext vs. injector.get()". So I hope it's fine to post my question here!
